workerapi/api.go

```diff
@@ -98,9 +98,19 @@ func ThriftRunStatusToDomain(thrift *worker.RunStatus) runner.RunStatus {
 	if thrift.ExitCode != nil {
 		domain.ExitCode = int(*thrift.ExitCode)
 	}
+	if thrift.SnapshotId != nil {
+		domain.SnapshotID = *thrift.SnapshotId
+	}
 	return domain
 }
 
+func copyString(s string) *string {
```

Comment:
It is important, because if we don't copy like this, we open ourselves to bugs more easily. (Because taking the address of a struct can mean your code changes out from under you if the struct is mutated)